core/src/main/java/org/apache/accumulo/core/rpc/ThriftUtil.java

```diff
@@ -53,12 +54,21 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Scheduler;
+
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 
 /**
  * Factory methods for creating Thrift client objects
  */
 public class ThriftUtil {
+
+  private static final Cache<SslConnectionParams,SSLContext> SSL_CONTEXT_CACHE =
+      Caffeine.newBuilder().scheduler(Scheduler.systemScheduler())
+          .expireAfterAccess(Duration.ofMinutes(10)).initialCapacity(1).maximumSize(8).build();
```

Comment:
I'm not sure a Map/Cache is needed here. If so, the expiration can certainly be longer and may help with TLS session resumption. My thought here is that there is *only* one SSLContext per client.

Reply:
The transport is already cached. I would expect a new SSLContext to be created for each new transport... we don't know if the underlying JCE library or some other state has changed that would result in a differently-constructed context for the same SSL params. If this is expensive, then perhaps we need to focus on caching the transport longer, rather than caching at both the SSLContext and the transport?